packages/api/src/middlewares/RouterMiddlewareFactory.ts

```diff
@@ -29,6 +29,77 @@ export function classMethodHandler(
     fn: method
   };
 }
+
+function getRouteRegex(path: string): RegExp {
+  const parts = path.split(/\{|\}/g);
+  let pattern = "";
+  const pathVariables: Array<{ name: string; value: any }> = [];
+  for (let i = 0; i < parts.length; i++) {
+    const isPathParam = i % 2 === 1;
+    if (!isPathParam) {
+      pattern += parts[i].replace(/\//g, "\\/");
+    } else {
+      pattern += "(.*?)";
+    }
+  }
+  return new RegExp(`^${pattern}$`);
+}
+
+function getPathVariables(
+  path: string
+): Array<{ name: string; value: string }> {
+  const parts = path.split(/\{|\}/g);
```

Comment:
This seems pretty hairy. Maybe we should make a Utils file for this factory to use.

Reply:
Also, document this, it's not intuitive at all and very terse.